### PR TITLE
chore: Adds Github Actions CI workflow

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

This PR introduces a minimal Github Actions CI workflow that validates:
- Unit tests
- The project builds

This change also includes a minimal fix in service/init.go that caused `go test` to fail.

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
```
Comment (gab-arrobo) [inline .github/workflows/master.yml:22]:
```suggestion
          go-version-file: 'go.mod'
```

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        run: go build
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
```
Comment (gab-arrobo) [inline .github/workflows/master.yml:34]: As per the [setup-go](https://github.com/marketplace/actions/setup-go-environment#getting-go-version-from-the-gomod-file)
```suggestion
          go-version-file: 'go.mod'
```

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
```
Comment (gab-arrobo) [inline .github/workflows/master.yml:2]: Is it 2023? or 2024?

File: service/init.go [hunk 4]
```diff
@@ -292,7 +292,7 @@ func (nssf *NSSF) BuildAndSendRegisterNFInstance() (models.NfProfile, error) {
 	return profile, err
 }
 
-//UpdateNF is the callback function, this is called when keepalivetimer elapsed
+// UpdateNF is the callback function, this is called when keepalivetimer elapsed
```
Comment (gab-arrobo) [inline service/init.go:295]: Is this a lint-related change?

File: .github/workflows/master.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
+
+      - name: Unit tests
+        run: go test ./...
```
Comment (gab-arrobo) [inline .github/workflows/master.yml:37]: What about adding a lint action?
```suggestion
        run: go test ./...

  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-go@v5
        with:
          go-version-file: 'go.mod'

      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3.7.0
        with:
          version: latest
          args: -v --config ./.golangci.yml

```

File: service/init.go (hunk 4 above)
Comment (gruyaume) [reply service/init.go:295]: Use a go lint (ctrl s) caused change

File: .github/workflows/master.yml (hunk 5 above)
Comment (gruyaume) [reply .github/workflows/master.yml:37]: Agree but let's do this in a different PR, it is independent

File: .github/workflows/master.yml (hunk 3 above)
Comment (gruyaume) [reply .github/workflows/master.yml]: Fixed

File: .github/workflows/master.yml (hunk 5 above)
Comment (gab-arrobo) [reply .github/workflows/master.yml:37]: ok, no problem!

**Review by gab-arrobo — APPROVED** [review]
+1

**Pull request merged**